src/table/components/head/HeadCellMenu.tsx

```diff
@@ -28,6 +41,18 @@ export default function HeadCellMenu({ column, tabIndex }: HeadCellMenuProps) {
     embed.on('fieldPopoverClose', () => {
       setOpenListboxDropdown(false);
     });
+  }, [embed, column.fieldId]);
+
+  useEffect(() => {
+    if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
+  }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
```

Comment:
Late to the party here, but i don't see why you need this. It doesn't matter what the statuses are when the menu is closed. We only need to make sure they are correct when you open the menu

Reply:
nvm now I remember. If they are in the wrong state if the getLayout is slow, they sholud be disabled and nothing else